lib/parser/create.js

```diff
@@ -315,21 +315,29 @@ export function createParser(config) {
         parser.parseValue = 'parseValue' in options ? Boolean(options.parseValue) : true;
         parser.parseCustomProperty = 'parseCustomProperty' in options ? Boolean(options.parseCustomProperty) : false;
 
-        const { context = 'default', onComment } = options;
+        const { context = 'default', onComment, onToken } = options;
 
         if (context in parser.context === false) {
             throw new Error('Unknown context `' + context + '`');
         }
 
-        if (typeof onComment === 'function') {
+        if (typeof onComment === 'function' || typeof onToken === 'function' || Array.isArray(onToken)) {
             parser.forEachToken((type, start, end) => {
                 if (type === Comment) {
-                    const loc = parser.getLocation(start, end);
-                    const value = cmpStr(source, end - 2, end, '*/')
-                        ? source.slice(start + 2, end - 2)
-                        : source.slice(start + 2, end);
+                    if (onComment) {
+                        const loc = parser.getLocation(start, end);
+                        const value = cmpStr(source, end - 2, end, '*/')
+                            ? source.slice(start + 2, end - 2)
+                            : source.slice(start + 2, end);
+
+                        onComment(value, loc);
+                    }
+                }
 
-                    onComment(value, loc);
+                if (typeof onToken === 'function') {
+                    onToken(type, start, end);
```

Comment:
I completely forgot! We need a 4th parameter, `index`. It's passed as the 4th argument in `forEachToken` and should be forwarded to `onToken`. This is important for determining the token's position and accessing values in other arrays like `balance`.